### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by overthemike**
## Related Bug Reports or Discussions
N/A

Fixes #
N/A

## Summary
* Formatting updates 
* Added [!TIP] and [!NOTE] and [!WARNING] where appropriate to highlight them a bit better
* Added diff block for migration to better see what needs to change
* Added internal links when talking about another section within a page

Preview: [https://github.com/overthemike/valtio/tree/main/docs](https://github.com/overthemike/valtio/tree/main/docs)

## Check List

- [ x ] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

File: docs/api/advanced/snapshot.mdx [hunk 1]
```diff
@@ -28,9 +28,9 @@ const snap3 = snapshot(store)
 console.log(snap1 === snap3) // false, should re-render
 ```
 
-## Copy on Write
+## <a name="copyonwrite"></a> Copy on Write
```
Comment (dai-shi) [inline docs/api/advanced/snapshot.mdx:31]: I think there's automatically generated link in github browser. (I'm not sure how it does with `website`.

```suggestion
## Copy on Write
```

File: docs/api/utils/devtools.mdx [hunk 2]
```diff
@@ -32,7 +32,7 @@ Valtio is not tied to React, you can use it in vanillaJS.
 ```jsx
 import { proxy, subscribe, snapshot } from 'valtio/vanilla'
 
-const state = proxy({ count: 0, text: 'hello' })
+const state = proxy({ count: 0, text: 'llo' })
```
Comment (dai-shi) [inline docs/api/utils/devtools.mdx:35]:
```suggestion
const state = proxy({ count: 0, text: 'hello' })
```

File: docs/guides/migrating-to-v2.mdx [hunk 3]
```diff
@@ -57,51 +57,37 @@ const Component = () => {
 }
 ```
 
+```diff
```
Comment (dai-shi) [inline docs/guides/migrating-to-v2.mdx:60]: You are overwriting the change in the main branch.

**Review by dai-shi — CHANGES_REQUESTED** [review]
quick review.

Please check vercel error too.

**Comment by dai-shi** [comment]
<details>
<summary>Vercel deploy error</summary>

```
[07:04:48.391] Running build in Washington, D.C., USA (East) – iad1
[07:04:48.502] Cloning github.com/pmndrs/valtio (Branch: docs, Commit: e60ac13)
[07:04:48.783] Cloning completed: 280.405ms
[07:04:52.298] Restored build cache from previous deployment (HrooWUGkUfhfeYhTRjQwJEdNiXBi)
[07:04:52.377] Running "vercel build"
[07:04:52.827] Vercel CLI 37.1.2
[07:04:53.238] Detected `pnpm-lock.yaml` version 6 generated by pnpm@8.x
[07:04:53.267] Installing dependencies...
[07:04:54.086] Lockfile is up to date, resolution step is skipped
[07:04:54.276] Already up to date
[07:04:54.871] 
[07:04:54.975] Done in 1.6s
[07:04:54.990] Detected Next.js version: 12.3.4
[07:04:55.009] Running "pnpm run build"
[07:04:55.513] 
[07:04:55.513] > website@ build /vercel/path0/website
[07:04:55.513] > next build
[07:04:55.514] 
[07:04:56.037] info  - Linting and checking validity of types...
[07:04:58.958] info  - Creating an optimized production build...
[07:05:06.882] 
[07:05:06.882] [1m[33mwarn[39m[22m - As of Tailwind CSS v2.2, `lightBlue` has been renamed to `sky`.
[07:05:06.882] [1m[33mwarn[39m[22m - Update your configuration file to silence this warning.
[07:05:06.882] 
[07:05:06.882] [1m[33mwarn[39m[22m - As of Tailwind CSS v3.0, `warmGray` has been renamed to `stone`.
[07:05:06.882] [1m[33mwarn[39m[22m - Update your configuration file to silence this warning.
[07:05:06.883] 
[07:05:06.883] [1m[33mwarn[39m[22m - As of Tailwind CSS v3.0, `trueGray` has been renamed to `neutral`.
[07:05:06.883] [1m[33mwarn[39m[22m - Update your configuration file to silence this warning.
[07:05:06.883] 
[07:05:06.883] [1m[33mwarn[39m[22m - As of Tailwind CSS v3.0, `coolGray` has been renamed to `gray`.
[07:05:06.883] [1m[33mwarn[39m[22m - Update your configuration file to silence this warning.
[07:05:06.883] 
[07:05:06.883] [1m[33mwarn[39m[22m - As of Tailwind CSS v3.0, `blueGray` has been renamed to `slate`.
[07:05:06.883] [1m[33mwarn[39m[22m - Update your configuration file to silence this warning.
[07:05:07.930] info  - Compiled successfully
[07:05:07.930] info  - Collecting page data...
[07:05:09.272] info  - Generating static pages (0/36)
[07:05:10.900] info  - Generating static pages (9/36)
[07:05:11.436] info  - Generating static pages (18/36)
[07:05:11.674] ✘ [ERROR] [plugin esbuild-xdm] Unexpected closing tag `</lu>`, expected corresponding closing tag for `<li>` (173:3-173:7)
[07:05:11.674] 
[07:05:11.674]     _mdx_bundler_entry_point-4eae3e99-c737-48d6-be62-35cea59be588.mdx:173:17:
[07:05:11.675]       173 │   <li>Update temp</lu>
[07:05:11.675]           ╵                  ~~~~~
[07:05:11.675] 
[07:05:11.675] 
[07:05:11.675] Error occurred prerendering page "/docs/how-tos/some-gotchas". Read more: https://nextjs.org/docs/messages/prerender-error
[07:05:11.676] Error: Build failed with 1 error:
[07:05:11.676] _mdx_bundler_entry_point-4eae3e99-c737-48d6-be62-35cea59be588.mdx:173:17: ERROR: [plugin: esbuild-xdm] Unexpected closing tag `</lu>`, expected corresponding closing tag for `<li>` (173:3-173:7)
[07:05:11.676]     at failureErrorWithLog (/vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:1624:15)
[07:05:11.676]     at /vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:1266:28
[07:05:11.676]     at runOnEndCallbacks (/vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:1046:63)
[07:05:11.676]     at buildResponseToResult (/vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:1264:7)
[07:05:11.676]     at /vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:1377:14
[07:05:11.676]     at /vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:678:9
[07:05:11.677]     at handleIncomingPacket (/vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:775:9)
[07:05:11.677]     at Socket.readFromStdout (/vercel/path0/website/node_modules/.pnpm/esbuild@0.14.54/node_modules/esbuild/lib/main.js:644:7)
[07:05:11.677]     at Socket.emit (node:events:517:28)
[07:05:11.677]     at addChunk (node:internal/streams/readable:368:12)
[07:05:11.732] info  - Generating static pages (27/36)
[07:05:11.927] info  - Generating static pages (36/36)
[07:05:11.929] 
[07:05:11.930] > Build error occurred
[07:05:11.932] Error: Export encountered errors on following paths:
[07:05:11.932] 	/docs/[...slug]: /docs/how-tos/some-gotchas
[07:05:11.932]     at /vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/export/index.js:404:19
[07:05:11.932]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[07:05:11.933]     at async Span.traceAsyncFn (/vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/trace/trace.js:79:20)
[07:05:11.933]     at async /vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/build/index.js:1230:21
[07:05:11.933]     at async Span.traceAsyncFn (/vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/trace/trace.js:79:20)
[07:05:11.933]     at async /vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/build/index.js:1092:17
[07:05:11.933]     at async Span.traceAsyncFn (/vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/trace/trace.js:79:20)
[07:05:11.933]     at async Object.build [as default] (/vercel/path0/website/node_modules/.pnpm/next@12.3.4_@babel+core@7.24.5_react-dom@17.0.2_react@17.0.2/node_modules/next/dist/build/index.js:65:29)
[07:05:11.961]  ELIFECYCLE  Command failed with exit code 1.
[07:05:11.982] Error: Command "pnpm run build" exited with 1
[07:05:12.394] 
```

</details>

File: docs/api/advanced/snapshot.mdx (hunk 1 above)
Comment (overthemike) [reply docs/api/advanced/snapshot.mdx]: Got it - changed it to the correct path and removed the anchor

File: docs/guides/migrating-to-v2.mdx (hunk 3 above)
Comment (overthemike) [reply docs/guides/migrating-to-v2.mdx:60]: Woops - not sure what happened there :\

**Comment by overthemike** [comment]
Fixed the error Vercel was complaining about.

File: docs/guides/migrating-to-v2.mdx [hunk 4]
```diff
@@ -102,6 +109,16 @@ state.obj = deepClone(newObj)
...
++ import { deepClone } from 'valtio/utils'     // v2
+
+- const state = proxy(initialObj)              // v1
++ const state = proxy(deepClone(initialObj))   // v2
+
+- state.obj = newObj                           // v1
++ state.obj = deepClone(newObj)                // v2
+```
+
```
Comment (dai-shi) [inline docs/guides/migrating-to-v2.mdx:121]: This is the confusing one, and #939 improved it.

```suggestion
```

File: docs/guides/migrating-to-v2.mdx [hunk 5]
```diff
@@ -57,6 +57,13 @@ const Component = () => {
 }
 ```
 
+```diff
+-    return <>{JSON.stringify(snap.data)}</>        // v1
++    return <>{JSON.stringify(use(snap.data))}</>   // v2
++    // If `data` is not an object, you can directly embed it in JSX.
++    // return <>{snap.data}</>
+```
+
```
Comment (dai-shi) [inline docs/guides/migrating-to-v2.mdx:66]: I think this is repetitive.

```suggestion
```

**Review by dai-shi — APPROVED** [review]
Looks good otherwise.

**Pull request merged**